metadb/leader_transfer.go

```diff
@@ -0,0 +1,135 @@
+package metadb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pastelnetwork/gonode/metadb/rqlite/store"
+
+	"github.com/pastelnetwork/gonode/common/errors"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/pastel"
+)
+
+// initLeadershipTransferTrigger is supposed to transfer leadership from current leader
+// to another Top Ranked SN every N blocks so the system remains decentralized
+func (s *service) initLeadershipTransferTrigger(ctx context.Context,
+	leaderCheckInterval time.Duration, blockCountCheckInterval time.Duration) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(leaderCheckInterval):
+			// this logic is only for leader, we don't want to exit for loop
+			// as current node can become leader in future
+			if !s.IsLeader() {
+				break
+			}
+
+			count, err := s.pastelClient.GetBlockCount(ctx)
+			if err != nil {
+				log.WithContext(ctx).WithError(err).Error("initLeaderElectionTrigger failed to call pastel client BlockCount")
+				break
```

Comment:
I think should retry again than break here?

Reply:
break only breaks switch case so it will try again after the interval (few seconds) ....so retrying here is not required

Reply:
Thanks.